README.md

```diff
@@ -841,6 +856,96 @@ Send the command to tell the board to start the syncing protocol. Must be connec
+
+**Note, this functionality requires OpenBCI Firmware Version 2.0**
+
+**_Returns_** {Promise} resolves if `synced` event is emitted, rejects if not connected or using firmware v2. Resolves with a synced object:
+```javascript
+{
+    boardTime: 0, // The time contained in the time sync set packet.
+    correctedTransmissionTime: false, // If the confirmation and the set packet arrive in the same serial flush we have big problem! This will be true in this case. See source code for full explanation.
+    timeSyncSent: 0, // The time the `<` was sent to the Dongle.
+    timeSyncSentConfirmation: 0, // The time the `<` was sent to the Board; It's really the time `,` was received from the Dongle.
+    timeSyncSetPacket: 0, // The time the set packet was received from the Board.
+    timeRoundTrip: 0, // Simply timeSyncSetPacket - timeSyncSent.
+    timeTransmission: 0, // Estimated time it took for time sync set packet to be sent from Board to Driver.
+    timeOffset: 0, // The map (or translation) from boardTime to module time.
+    valid: false // If there was an error in the process, valid will be false and no time sync was done. It's important to resolve this so we can perform multiple promise syncs as show in the example below.
+}
+```
+
+**Example**
+
+```javascript
+var OpenBCIBoard = require('openbci').OpenBCIBoard,
+    ourBoard = new OpenBCIBoard({
+        verbose:true
+    });
+
+var portName = /* INSERT PORT NAME HERE */;
+var samples = []; // Array to store time synced samples into
+var timeSyncActivated = false;
+
+ourBoard.connect(portName)
+    .then(() => {
+        ourBoard.on('ready',() => {
+            ourBoard.streamStart()
+                .then(() => {
+                    /** Could also call `.syncClocksFull()` here */
+                })
+                .catch(err => {
+                    console.log(`Error starting stream ${err}`);
+                })
+            });
+        ourBoard.on('sample',sample => {
+            /** If we are not synced, then do that! */
+            if (timeSyncActivated === false) {
+                timeSyncActivated = true;
+                ourBoard.syncClocksFull()
+                    .then(syncObj => {
+                        if (syncObj.valid) {
+                            console.log('1st sync done');
+                        }
+                        return ourBoard.syncClocksFull();
+                    })
+                    .then(syncObj => {
+                        if (syncObj.valid) {
+                            console.log('2nd sync done');
+                        }
+                        return ourBoard.syncClocksFull();
+                    })
+                    .then(syncObj => {
+                        if (syncObj.valid) {
+                            console.log('3rd sync done');
+                        }
+                        return ourBoard.syncClocksFull();
+                    })
+                    .then(syncObj => {
```

Comment:
is this just demonstrating how to do four time syncs? the example above had the sync based on the modulus of the sample count, which makes sense.